### PR TITLE
feat(test): Add UBAClientUtilities unit tests

### PR DESCRIPTION
**Pull request by nicholaspai**
Companion unit tests to SDK PR [#308](https://github.com/across-protocol/sdk-v2/pull/308/files#diff-d2decf35e36422291323a94ed96a949778b031996571644c1b65d3ffcf40db4e)

### CODE REVIEW

File: test/UBAFeeConfig.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+let originToken: string;
+let config: MockUBAConfig;
+
+const originChainId = 10;
+const destinationChainId = 137;
+const hubPoolChainId = constants.HUBPOOL_CHAIN_ID;
+
+describe("UBAFeeConfig", async function () {
+  describe("getTotalSpokeTargetBalanceForComputingLpFee", function () {
```
Comment (nicholaspai) [inline test/UBAFeeConfig.ts:14]: This function is now not used in the UBAFeeConfig but effectively tests internal functions like `getBalanceTriggerThreshold`

File: test/Relayer.BasicFill.ts [hunk 2]
```diff
@@ -423,8 +423,31 @@ describe("Relayer: Check for Unfilled Deposits and Fill", async function () {
...
+        inventoryClient: new MockInventoryClient(),
+        acrossApiClient: new AcrossApiClient(spyLogger, hubPoolClient, spokePoolClients),
+      },
+      {
+        relayerTokens: [],
+        relayerDestinationChains: [originChainId, destinationChainId],
+        minDepositConfirmations: defaultMinDepositConfirmations,
+        quoteTimeBuffer: 0,
+      } as unknown as RelayerConfig
+    );
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply test/Relayer.BasicFill.ts]: https://github.com/across-protocol/relayer-v2/commit/2f3f935304c234a9a9533f201e6b27de67e1eb42

File: test/Relayer.BasicFill.ts [hunk 3]
```diff
@@ -423,8 +423,31 @@ describe("Relayer: Check for Unfilled Deposits and Fill", async function () {
...
+        multiCallerClient,
+        inventoryClient: new MockInventoryClient(),
+        acrossApiClient: new AcrossApiClient(spyLogger, hubPoolClient, spokePoolClients),
+      },
+      {
+        relayerTokens: [],
+        relayerDestinationChains: [originChainId, destinationChainId],
+        minDepositConfirmations: defaultMinDepositConfirmations,
+        quoteTimeBuffer: 0,
+      } as unknown as RelayerConfig
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply test/Relayer.BasicFill.ts]: yep

File: test/Relayer.BasicFill.ts [hunk 4]
```diff
@@ -421,16 +423,14 @@ describe("Relayer: Check for Unfilled Deposits and Fill", async function () {
     expect(lastSpyLogIncludes(spy, "Skipping deposit for unsupported destination chain")).to.be.true;
   });
 
-  it("UBA: Ignores deposits after version bump", async function () {
+  it("UBA: Crashes if client cannot support version bump", async function () {
+    // Client is out of sync with on chain version, should crash.
     await configStore.updateGlobalConfig(utf8ToHex("VERSION"), `${UBA_MIN_CONFIG_STORE_VERSION ?? 2}`);
     // "reasonable" block number based off the block time when looking at quote timestamps.
     await spokePool_1.setCurrentTime(await getLastBlockTime(spokePool_1.provider));
     await deposit(spokePool_1, erc20_1, depositor, depositor, destinationChainId);
```
Comment (pxrl) [inline test/Relayer.BasicFill.ts:431]: I think this is unnecessary now.

```suggestion
```

File: test/Relayer.BasicFill.ts [hunk 5]
```diff
@@ -421,16 +423,14 @@ describe("Relayer: Check for Unfilled Deposits and Fill", async function () {
...
 
-    await updateAllClients();
-    await relayerInstance.checkForUnfilledDepositsAndFill();
-    expect(lastSpyLogIncludes(spy, "No unfilled deposits")).to.be.true;
-    expect(multiCallerClient.transactionCount()).to.equal(0); // One transaction, filling the one deposit.
+    await assertPromiseError(updateAllClients(), "ConfigStoreClient cannot handle UBA config store version");
```
Comment (pxrl) [inline test/Relayer.BasicFill.ts:433]:
```suggestion
    await assertPromiseError(configStoreClient.update(), "ConfigStoreClient cannot handle UBA config store version");
```

**Pull request merged**